src/video/haiku/SDL_bwindow.cc

```diff
@@ -127,8 +127,9 @@ void HAIKU_SetWindowTitle(_THIS, SDL_Window * window) {
     _ToBeWin(window)->PostMessage(&msg);
 }
 
-void HAIKU_SetWindowIcon(_THIS, SDL_Window * window, SDL_Surface * icon) {
-    /* FIXME: Icons not supported by Haiku */
+int HAIKU_SetWindowIcon(_THIS, SDL_Window * window, SDL_Surface * icon) {
+    /* Icons not supported by Haiku */
```

Comment:
This whole function can be removed. We now return SDL_Unsupported() if there is no driver implementation.

Reply:
So both the definition and the declaration can be removed for drivers that do not support setting a window's icon? @slouken

Reply:
Yes